src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -262,6 +263,16 @@ public async Task<bool> TryDownloadEmbeddedFile(string pluginKey, string embedde
             }, url);
         }
 
+        public Task<AnalysisCacheMsg> DownloadCache(string projectKey, string branch)
+        {
+            _ = projectKey ?? throw new ArgumentNullException(nameof(projectKey));
+            _ = branch ?? throw new ArgumentNullException(nameof(branch));
+
+            logger.LogDebug(Resources.MSG_DownloadingCache, projectKey, branch);
+            var url = GetUrl("/api/analysis_cache/get?project={0}&branch={1}", projectKey, branch);
```

Comment:
For `projectKey`, we're sure that it's ascii-only, right? 

We're missing `UrlEncode` for `branch`. Branch name often contains at least `/` from URL non-friendly characters. We need some kind of UT for that too.

Reply:
In general, this class is missing a UrlEncode for every single parameter. As a good practice, it should be everywhere - a thing to consider for the cleanup PR.